dapr/clients/grpc/client.py

```diff
@@ -984,6 +987,99 @@ def unsubscribe_configuration(
+
+            Finally, you can use this response with an with statement, and have the lock
+            be automatically unlocked after the with-statement scope ends
+
+                with client.try_lock(store_name, resource_id, first_client_id, expiry_s) as lock:
+                    if lock:
+                        # lock acquired successfully...
+                # Lock automatically unlocked at this point, no need to call client->unlock(...)
+
+            Args:
+                store_name (str): the lock store name, e.g. `redis`.
+                resource_id (str): the lock key. e.g. `order_id_111`.
+                                    It stands for "which resource I want to protect".
+                lock_owner (str):  indicates the identifier of lock owner.
+                expiry_in_seconds (int): The length of time (in seconds) for which this lock
+                    will be held and after which it expires.
+                metadata (tuple, optional, DEPRECATED): gRPC custom metadata
+
+            Returns:
+                :class:`TryLockResponse`: With the result of the try-lock operation.
+        """
+        warn('The Distributed Lock API is an Alpha version and is subject to change.',
+             UserWarning, stacklevel=2)
+        if metadata is not None:
+            warn('metadata argument is deprecated. Dapr already intercepts API token headers '
+                 'and this is not needed.', DeprecationWarning, stacklevel=2)
+        req = api_v1.TryLockRequest(
+            store_name=store_name,
+            resource_id=resource_id,
+            lock_owner=lock_owner,
+            expiryInSeconds=expiry_in_seconds)
+        response, call = self._stub.TryLockAlpha1.with_call(req, metadata=metadata)
+        return TryLockResponse(
+            success=response.success,
+            client=self,
+            store_name=store_name,
+            resource_id=resource_id,
+            lock_owner=lock_owner,
+            headers=call.initial_metadata())
+
+    def unlock(
+            self,
+            store_name: str,
+            resource_id: str,
+            lock_owner: str,
+            metadata: Optional[MetadataTuple] = None) -> UnlockResponse:
+        """Unlocks a lock.
+
+            Args:
+                store_name (str): the lock store name, e.g. `redis`.
+                resource_id (str): the lock key. e.g. `order_id_111`.
+                                    It stands for "which resource I want to protect".
+                lock_owner (str):  indicates the identifier of lock owner.
+                metadata (tuple, optional, DEPRECATED): gRPC custom metadata
+
+            Returns:
+                :class:`UnlockResponseStatus`: Status of the request,
+                    `UnlockResponseStatus.success` if it was successful of some other
+                    status otherwise.
+        """
+        warn('The Distributed Lock API is an Alpha version and is subject to change.',
```

Comment:
pending if I should do this to all methods.

While I was coding I considered creating a decorator for issuing these warning, I guess it would "document" the "alpha-ness" of these methods better. ;) Thoughts?